### PR TITLE
MsAI - New parachain for health

### PR DESCRIPTION
**Pull request by salmanmarvasti**
### Project Abstract

> Please provide a brief description of your project here summarising key points (1-2 paragraphs).
We have designed a health app which is need of a decentralised storage where we think web 3 technologies can really help.
 Integrating application specific security and storage for every application is a major pain point which is not addressed by blockchains.
We propose to create a blockchain Pallet and customise a Substrate chain such that data storage and encryption would be provided through an auditable blockchain. Something like the w3c Solid pod decentralised storage integrated within a Susbtrate node.

> If your application is a follow-up to a previous grant, please mention which one in the first line of the abstract and include a link to previous pull requests if applicable.
NA
### For which grant [level](https://github.com/w3f/Grants-Program#level_slider-levels) are you applying?

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [ x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for > $100k Web3 Foundation Council approval)

### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( `project_name.md`) and updated.
- [x] I have read and understood the [FAQs](https://github.com/w3f/Grants-Program/blob/master/docs/faq.md), [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/grant_guidelines_per_category.md) and [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/announcement-guidelines.md).
- [x] A BTC, Ethereum (USDT/USDC/DAI) or Polkadot/Kusama (aUSD) address for the payment of the milestones is provided inside the application.
- [x] I have read and acknowledge the [terms and conditions](https://github.com/w3f/Grants-Program/blob/master/docs/T&Cs.md).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process).
- [x] I prefer the discussion of this application to be in a private Element/Matrix channel. My username is: ``@_______``

### How Did You Hear About our grants program?

- [ ] Social Media
- [x] Hackathon
- [ ] Personal Recommendation
- [ ] Substrate Builders Program
- [ ] Investor/VC
- [ x] Online Search
- [ ] Other: _______

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the application. However, the formatting of your milestone delivery templates currently has some issues. Additionally each milestone should have the default deliveries of our template (Deliverables 0a-0d). And are you aware of offchain-worker: https://docs.substrate.io/reference/how-to-guides/offchain-workers/?

**Comment by salmanmarvasti** [comment]
Thanks will fix deliverables when we get time.
Btw off chain storage however doesn't provide any security or other
services needed for health data storage including distribution. I will look
into it if it provides any of the other features in a easy to use manner.

**Comment by salmanmarvasti** [comment]
I signed the CLA but it doesnt seem to have updated ?

**Comment by Noc2** [comment]
> SalMarvasti

Every contributor needs to sign the CLA, currently two github accounts seem to have contributed: 

- https://github.com/SalMarvasti 
- https://github.com/salmanmarvasti

**Comment by salmanmarvasti** [comment]
The table formats are fixed, any other changes ? I see it still is labelled changes requested.

**Comment by salmanmarvasti** [comment]
These are all a work in progress

**Comment by salmanmarvasti** [comment]
OFW is the off chain worker is the one you mentioned. According to Substrate docs already allows off chain storage in a basic way.

**Comment by Noc2** [comment]
> * u need to develop as part of the grant? Our grants program focuses on funding open source software development and research that is potentially useful for our ecosystem, doesn’t exist already and can be reused by others. We don’t support the deployment of existing software or reusing this software (e.g.: “Basic Blockchain parachain build using current available Pallets”).

That’s correct, see https://docs.substrate.io/reference/how-to-guides/offchain-workers/offchain-local-storage/ But the abbreviation is OCW as far as I know. That’s why I was confused.

**Comment by Noc2** [comment]
> These are all a work in progress

So, you are planning to update the application, correct?

File: applications/msai.md [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+* Decentralised: Data ownership is not lost to any central authority and access is controlled through secure ownership of NFTs
+* Incentive to participate: various mechanisms  could be developed as smart tokens or native token of the parachain linked to DOT.
+* Users:  publishers of health can receive some amount of token as rewards from any sponsor on the platform. 
+* Public: The api and parachain would be public so that any other apps such that deal with medical data can use this parachain. 
+* The detailed API and openAPI specifications for GRPC and or REST will be developed as part of this proposal with reference to the Solid pod api as details depends on the public blockchain that the system is implemented in.
+### Project Details
+
+Data models
+
+We propose to store data encrypted at client side using the Pod api . The data would be anonymous at the storage layer and encrypted with access to ID available only to owner(s) of the NFT controlled via a smart contract.
```
Comment (semuelle) [inline applications/msai.md:50]: Can you explain how ownership of an NFT could enable someone to decrypt data?

File: applications/msai.md [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+
+ID verification and data storage could be integrated through a Sidecar approach ( a containerized system design pattern) allowing other app developers within the ecosystem to reuse the chain without needing their own backend. 
+For a simpler initial alpha we could integrate IPFS as a pod making it simpler than the full Pod api [4].
+Distributed data could be verified as authentic by having a secure hash stored on the chain with the data in the separate pod.
+With trust being established by the blockchain layer, any blockchain node that provides faulty data as verified through smart contract hashes will be kicked out and forfiet their staking in the parachains native token.
+We also propose specific data format for anonymous tags such that no user identified private data would be exposed
+during the whole process. Additional layers of encryption and anonymity would be provided in subsequent stages.
+
+The initial user of this parachain would be our own app where users store any medical documents with automated information extraction, along with authenticated GPDR opt out.
```
Comment (semuelle) [inline applications/msai.md:32]: How do you extract information if the data is encrypted client-side?

**Review by semuelle — CHANGES_REQUESTED** [review]
Thanks for the application, @salmanmarvasti. I have a couple of questions.

Also, could you clarify what the intention and intended audience of the project is? At the top, you write "Many patients use clinics worldwide, but they face difficulty moving patient records" and "It is inconvenient and impractical to carry data folders over particularly between countries." Later you write "[...] would allow game theory incentives and smart contract based marketplaces for data trading" and "target audience is any healthcare data silo and any health patients who can contribute their data and earn ERC20 like tokens".

These are two quite different objectives. If this is about my most important medical data, I wouldn't want access to it to be sold or transferred. If this is about making money off my activity tracker data, then I'm not sure such levels of security and incentivisation are necessary.

**Comment by semuelle** [comment]
Also, please host the images elsewhere. For example, if you copy and paste them into the Github editor, it'll automatically upload them to githubusercontent.com.

File: applications/msai.md (hunk 1 above)
Comment (salmanmarvasti) [reply applications/msai.md]: I added some explanation it is by mechanism of the added logic to the offchain worker storage api that is specialised for this application.

File: applications/msai.md (hunk 2 above)
Comment (salmanmarvasti) [reply applications/msai.md]: We extract information with the clients permission when they opt in and use our application. We cannot do this if they use another app.

**Comment by salmanmarvasti** [comment]
I have made changes can you take a look. thanks. Please be aware as part of the proposal the details will be worked out when we actually extend the off chain storage api. As the details depend on the features of the blochchain we decide to use for our application

**Comment by salmanmarvasti** [comment]
> Thanks for the update. The label needs to be manually updated and isn’t changed automatically. So it might take some time until someone takes another look at the application. I have a couple of additional questions/comments:
> 
> * In general, for the deliveries could you only focus on the new things that you need to develop as part of the grant? Our grants program focuses on funding open source software development and research that is potentially useful for our ecosystem, doesn’t exist already and can be reused by others. We don’t support the deployment of existing software or reusing this software (e.g.: “Basic Blockchain parachain build using current available Pallets”).
> * What do you mean with OFW?
> * How exactly do you ensure data privacy?
> * The total costs are currently not the sum of the payments for the milestones.
> * Unrelated to the grant itself, but your website https://www.marvsai.com/ still has a lot of issues (e.g. links don’t work) and shows the text from the template (e.g. “Ready to use HTML/CSS device mockups, no Photoshop required!”).

we have updated and fixed some isssues  with the website.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update. However, the total costs are currently still not the sum of the milestones. The second milestone doesn’t have the default deliverables (0a to 0d), no associated costs and something like “Kusama Reference Testing” is hard for us to evaluate. Could you focus on the development work that you would do as part of the milestone? Also you still need to sign the Terms and Conditions (see above).

**Comment by salmanmarvasti** [comment]
Can they not be included in one milestone?

**Comment by Noc2** [comment]
What exactly do you mean? In case you are referring to the default deliverables, they should be part of every milestone, since each deliverable should be open source software, well documented and tested.

**Comment by salmanmarvasti** [comment]
A few questions, how do I sign the modified CLA? the page doesnt work anymore I had signed the original version.
The two milestones are now separately addresses and I believe the total matches Tier 2 Unless you changed it again from 30K

**Comment by salmanmarvasti** [comment]
@Noc2 We moved images to our own website host . any further changes?

File: applications/msai.md [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+- **Payment Address:** 0x8f1013fa606c6fcbcd3eff057e5b320b0c5f72e2
+- **[Level](https://github.com/w3f/Grants-Program/tree/master#level_slider-levels):** 2
+
+> ⚠️ *The combination of your GitHub account submitting the application and the payment address above will be your unique identifier during the program. Please keep them safe.*
+
+## Project Overview
+### Overview
+Many patients use clinics worldwide, but they face difficulty moving patient records, thus they have to take redundant duplicate tests. It is inconvenient and impractical to carry data folders over particularly between countries.  
+To solve this problem we gathered a team and designed an international app. We wanted to store the data in a decentralised manner.
+For further background you can[ read our blog](https://www.notion.so/Why-we-need-a-blockchain-for-health-data-1080c5b727964721822650972c28cab2)
```
[the remark this answers is not in the document]
Comment (salmanmarvasti) [reply applications/msai.md]: i made document public, also the users are Medical clinics like the ones we have contracted with in Washington DC ARE but  the blockchain based changes are public for everyone including rival apps to use securely and with publicly verifiable hashes through an easy to use API.  Also see nature  reference at end of proposal

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update. For me it looks like you used three different accounts to create this PR (​​ salmanmarvasti, Salman, SalMarvasti) and only one signed the terms and conditions. Worst case you could close this application and simply reopen a new one to resolve this. The PR is still adding a bunch of umls and one png to an img folder. Regarding your second milestone, could you remove “Kusama Reference Deployment”, we don’t support the deployment on kusama or any other deployment via our grants program. Regarding the example application, could you provide more details, like for example the programming language, is it an UI or a CLI?, etc. as part of the milestone specification. Also the specification of the documentation, should be something like this: “We will provide both inline documentation of the code and a basic tutorial that explains how a user can (for example) spin up one of our Substrate nodes and send test transactions, which will show how the new functionality works.” and not a link to a notion document (that could be changed without our agreement). The specification of the milestone are the requirements of our contracts.

**Comment by SalMarvasti** [comment]
I created a new branch and pull request. https://github.com/w3f/Grants-Program/pull/1046

**Comment by SalMarvasti** [comment]
Feel free to close this when you have time to look at the other branch.

**Comment by salmanmarvasti** [comment]
closing to see if other is fixed.

**Comment by Noc2** [comment]
I’m closing this application, because of your new PR: https://github.com/w3f/Grants-Program/pull/1141

**Pull request closed without merging**